batch/batch/front_end/front_end.py

```diff
@@ -1018,6 +1020,7 @@ async def _create_jobs(userdata: dict, job_specs: dict, batch_id: int, update_id
                 inst_coll_name,
                 n_regions,
                 regions_bits_rep,
+                run_condition == 'any' and len(parent_ids) > 0,
```

Comment:
This took me a couple reads to get what's going on here and this is a clever way to get at the state change of `any` jobs. However, it feels pretty funky to me that `any` jobs (mostly) start out as cancelled and are switch to not-cancelled when a parent succeeds. It feels like there should be a distinction between the states

- "none of my parents have completed yet"
- "all my parents so far have failed"
- "I will never run because all my parents failed"

I also don't love that `n_cancelled` can potentially go down and that a job could get uncancelled (what if I hit the `cancel` button which sets the job as cancelled but then a succeeding parent uncancels it?

How much work would it be to do the following?

All jobs start as not cancelled. We track `n_succeeded_parents`, and then we can set an any job as cancelled if `n_parents > 0 & n_pending_parents = 0 & n_succeeded_parents = 0`?